### ISSUE TITLE
One initial condition for mcm, not working

### ISSUE THREAD

**Issue by elBuffer** [JR-1991/Catalax#16]
Hey! Awesome work, something that i have been looking for a long time.

I was trying to run this 
# Define Priors
model.parameters.k_cat.prior = cmc.priors.TruncatedNormal(mu=10.0, sigma=5.0, low=5.0, high=30.0)
model.parameters.K_m.prior = cmc.priors.TruncatedNormal(mu=100.0, sigma=10.0, low=10.0, high=1000.0)
model.parameters.k_d.prior = cmc.priors.Uniform(low=1e-5, high=9e-3)

mcmc, _ = cmc.run_mcmc(
    model=model,
    data=data,
    initial_conditions=initial_conditions,
    times=jnp.stack([time]*len(initial_conditions), axis=0),
    yerrs=0.97,
    num_warmup=1000,
    num_samples=1000,
    dt0=1.0,
    dense_mass=True,
    max_steps=63**4,
)

for just one condition  (initial_conditions)[0]

however, the shaping isn't happening as it should, so the y0s, shapes to (2,) instead of (1,2) as it should , this only happens for this one case, if the initial condition >1 then it works well

**Comment by JR-1991** [comment]
@elBuffer Thanks, that is great to hear! Also, thanks for submitting this issue. The bug regarding single datasets is known, and I am currently in the midst of fixing this, plus adding a smoother integration for data. The fix will be up soon and you can follow it in this [branch](https://github.com/JR-1991/Catalax/tree/data-input-class).

I will keep you posted 🙌

**Comment by JR-1991** [comment]
@elBuffer, I apologize for the delay, but the issue has been resolved by adding numerous other features to the [Dataset](https://github.com/JR-1991/Catalax/blob/master/catalax/dataset/dataset.py). You can refer to this [notebook](https://github.com/JR-1991/Catalax/blob/master/examples/GenerateData.ipynb) and most other examples in the [examples](https://github.com/JR-1991/Catalax/tree/master/examples) directory for illustrative usage.
